cache.go

```diff
@@ -87,15 +90,37 @@ func (cd *cachedData) flush() {
 		return
 	}
 
-	file, err := os.OpenFile(cd.Path, os.O_RDWR|os.O_TRUNC, 0666)
+	file, err := fileutil.LockFile(cd.Path, os.O_RDWR, 0666)
 	if err != nil {
 		log.Printf("Cannot write to the cache file: %v", err)
```

Comment:
I think that just a `"cannot write file"` could be interpreted in the wrong way. Maybe we could give a more descriptive error when it is because of a concurrency problem. Something like` "... some other instance if using the file..."` or something like that...